### PR TITLE
[Documentation] Setup-google-maps.md

### PR DESCRIPTION
**Pull request by richardbluestone**
## Status
Ready/In Progress/In Hold (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

### CODE REVIEW

**Comment by richardbluestone** [comment]
@altmannyarden Not sure if you need to approve here, but you can review if required.

File: content-repo/extra-docs/articles/setup-google-maps.md
Comment (ali-sawyer) [inline content-repo/extra-docs/articles/setup-google-maps.md:7]: Use consistent font for both the `ShowOnMap` and `ShowLocationOnMap` automations in the first sentence.

Comment (ali-sawyer) [reply content-repo/extra-docs/articles/setup-google-maps.md:7]: You need to create a Google Cloud Platform project, not create an API. Then you need to enable the Maps JavaScript API and create an API key for the project.

File: content-repo/extra-docs/articles/setup-google-maps.md
Comment (ali-sawyer) [inline content-repo/extra-docs/articles/setup-google-maps.md:13]: Recommend calling the API only by the name Google uses in its documentation, Maps JavaScript API, to avoid confusion.

File: content-repo/extra-docs/articles/setup-google-maps.md
Comment (ali-sawyer) [inline content-repo/extra-docs/articles/setup-google-maps.md:25]: Again here, recommend changing to "Maps JavaScript API". Also double check the step number.

File: content-repo/extra-docs/articles/setup-google-maps.md
Comment (ali-sawyer) [inline content-repo/extra-docs/articles/setup-google-maps.md:28]: Literal `s are showing up in this line rather than the different font, for some reason.

File: content-repo/extra-docs/articles/setup-google-maps.md
Comment (ali-sawyer) [inline content-repo/extra-docs/articles/setup-google-maps.md:35]: Double check the image – it is not showing up in the preview (same with the other images); not sure if that is expected.

File: content-repo/extra-docs/articles/setup-google-maps.md
Comment (ali-sawyer) [inline content-repo/extra-docs/articles/setup-google-maps.md:45]: The field name has a space in it: Geo Location

Change last sentence to: "The map should be shown in the section that you created."

File: content-repo/extra-docs/articles/setup-google-maps.md
Comment (ali-sawyer) [inline content-repo/extra-docs/articles/setup-google-maps.md:3]: Change to something like: "Setup Google Maps in Cortex XSOAR to Use Map Automations"

Otherwise, it makes it sound like you should use an automation to perform the setup, which is not the case.

**Review by ali-sawyer — COMMENTED** [review]
Added a few comments. Thanks for getting this done, @richardbluestone !

**Review by richardbluestone — COMMENTED** [review]
Made the changes

**Comment by altmannyarden** [comment]
I dont understand - If i have the Google Maps integration enabled, do I still need to do those steps?

**Comment by richardbluestone** [comment]
> I dont understand - If i have the Google Maps integration enabled, do I still need to do those steps?

As far as I understand, for both automations and the integration you need to add the  API by following the steps in GCS.
To run the automations,  you need to enter the server config and for the `ShowLocationOnMap` automation you also need to add the automation to a layout.  For the integration you don't need the server config for it to work. See https://github.com/demisto/etc/issues/45220#issue-1090284655

**Comment by altmannyarden** [comment]
I saw the issue and the KB, I still think it's a not a best practice to do add the API key in the server configuration.
Will update here after checking this issue. @richardbluestone

**Comment by ali-sawyer** [comment]
From my testing, adding the `ui.google.api.key` server config was the only way I could get these automations to work. The Google Maps integration is separate and unrelated (it uses a different Google API, the Geocoding API). This could be confusing for users, but that is a bigger issue.

**Comment by richardbluestone** [comment]
@altmannyarden @ali-sawyer What do we want to do about the server configs? Once approved I can then update https://github.com/demisto/content/pull/16923

**Comment by altmannyarden** [comment]
>From my testing, adding the ui.google.api.key server config was the only way I could get these automations to work. The Google Maps integration is separate and unrelated (it uses a different Google API, the Geocoding API). This could be confusing for users, but that is a bigger issue.

How it is different? The showonmap script using the command from the Google Maps integration. 

Anyway, I tested the "showonmap" automation:
1. If I configured the Google Maps integration -  the automation works.
2. If I added only the API key using the server configuration - the automation **failed**. 

I wanted to test the showlocation script, but I couldn't find any integration that map the "geolocation" field.
@ali-sawyer @richardbluestone

**Comment by richardbluestone** [comment]
@altmannyarden To see the `Geo Location field, go to **Settings>Objects Setup>Fields** and search for **Geo Location**. Edit the field and select **Add to all Indicator types**. When you edit the indicator, you should see this field. Of course you need to add the general purpose dynamic section and select the showlocation script to the layout and then add the layout to an indicator type. I will add this as note about the Geo Location field, as its not clear enough.
For the showonmap script it didn't make any difference with the integration enabled or the server config. The map appeared but with an error message.
For the showlocation script it only worked without the server config but with the same error (see attached).
I think its because billing is not enabled in GCS. We have to enter the API key either in the integration or server config to remove the error message. @ali-sawyer Do you have an API I can test?
image

**Comment by richardbluestone** [comment]
I fired up a new instance without any integrations or server configs. You can run both scripts without the server config or instance, but the map will appear with an error. When entering the API key, the error is removed.
image

File: content-repo/extra-docs/articles/setup-google-maps.md
Comment (ali-sawyer) [inline content-repo/extra-docs/articles/setup-google-maps.md:7]: Same automation name `ShowLocationOnMap` repeated in first sentence. One should be `ShowOnMap`.

File: content-repo/extra-docs/articles/setup-google-maps.md
Comment (ali-sawyer) [inline content-repo/extra-docs/articles/setup-google-maps.md:13]: Missing space before API.

File: content-repo/extra-docs/articles/setup-google-maps.md
Comment (ali-sawyer) [inline content-repo/extra-docs/articles/setup-google-maps.md:28]: Suggest changing to "Drag and drop"

File: content-repo/extra-docs/articles/setup-google-maps.md
Comment (ali-sawyer) [inline content-repo/extra-docs/articles/setup-google-maps.md:30]: Not a big deal! But in this example, it is called: `General Purpose Dynamic Section - ShowLocationOnMap`

**Review by ali-sawyer — APPROVED** [review]
Called out a couple typos, but they are small and quick fixes. Happy to re-approve if needed after you have made the changes, just let me know. Very exciting to see this close to done, I think this will be a helpful resource!

**Pull request merged**
